### PR TITLE
Dead end the Microsoft.Extensions.Configuration.NewtonsoftJson package

### PR DESCRIPTION
**Pull request by Anipik**
Fixes https://github.com/dotnet/extensions/issues/3816

- removing the  Microsoft.Extensions.Configuration.NewtonsoftJson package
- updating the preview label 3 (just for consistency, we wont be shipping any packages from here)
- updating the readme file
- It doesnt seem like we are using this package anywhere in our build.

### CODE REVIEW

**Comment by ViktorHofer** [comment]
Just to make sure, did you get sign-off from the library owner's to remove it?

**Review by ViktorHofer — APPROVED** [review]

**Comment by Anipik** [comment]
> Just to make sure, did you get sign-off from the library owner's to remove it?

I think runtime team is supposed to own this package and i dont think we have any person assigned as owner for this one.

**Review by eerhardt — APPROVED** [review]

**Comment by davidfowl** [comment]
Why are we doing this?

**Comment by eerhardt** [comment]
@davidfowl - more reasoning than what is listed in #3816:

1. This is the only remaining "shipping" package in dotnet/extensions. This will allow us to no longer build/maintain the dotnet/extensions repo in the main branch.
2. We moved the rest of the packages either to dotnet/aspnetcore or dotnet/runtime. Moving this one to dotnet/runtime will make a "product dependency" from dotnet/runtime to Newtonsoft.Json, which isn't ideal. dotnet/rutnime is supposed to be the "bottom of the stack".

So unless we can find another home for this package (which only has 42k total downloads in 2 years), it doesn't warrant the maintenance burden.

Is there some unique benefit this package provides that https://www.nuget.org/packages/Microsoft.Extensions.Configuration.Json/ doesn't?

File: eng/Versions.props [hunk 1]
```diff
@@ -11,7 +11,7 @@
     <PatchVersion>0</PatchVersion>
     <VersionPrefix>$(MajorVersion).$(MinorVersion).$(PatchVersion)</VersionPrefix>
     <PreReleaseVersionLabel>preview</PreReleaseVersionLabel>
-    <PreReleaseVersionIteration>2</PreReleaseVersionIteration>
+    <PreReleaseVersionIteration>3</PreReleaseVersionIteration>
```
Comment (mmitche) [inline eng/Versions.props:14]: Why is this getting mvoed to 3 now (vs. when we branch?)

Comment (Anipik) [reply eng/Versions.props:14]: I just combined the changes together, we wont be building or shipping anything from this in preview 2.

Comment (mmitche) [reply eng/Versions.props:14]: Is it being retired going forward? Or will there be things for preview 3?

Comment (Anipik) [reply eng/Versions.props:14]: its being retired going forward. we dont want to ship anything in preview2 as well from extensions repo.

**Comment by davidfowl** [comment]
Got it, if we don't see the value nor the usage then I'm ok with it.

**Pull request merged**